### PR TITLE
AWS lambda and tests

### PR DESCRIPTION
**Pull request by anya-mb**
## Description 

The plugin was updated to do text processing on AWS infrastructure using AWS lambda and tested with pytest.

#### infra/infra_stack.py
Added main AWS logic for lambda, HTTP API and routes

#### infra/app.py, infra/cdk.json
Initialized app

#### infra/pyproject.toml
Added Poetry for dependency management for infra too

#### backend/main.py
Replaced FastAPI and pydantic with lambda function

#### backend/pyproject.toml
Granted plugin permission to call endpoint on AWS 

#### backend/poetry.lock
Deleted unused dependencies (FastAPI and pydantic), added pytest

#### tests/test_reverse_text_work_correctly.py
Added a simple and a parametrized tests for backend

#### README.rst
Updated README

## Testing

Confirmed that the plugin reverses the text in Google Chrome browser.

Confirmed that the plugin reverses the text in Postman and returns an operation as well:
image

Confirmed that tests passed:
image

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -12,7 +12,14 @@ Directory Structure
     │   ├── main.py
     │   ├── pyproject.toml
     │   └── poetry.lock
-    │
+    ├── infra
+    │   ├── app.py
```
Comment (anatolyburtsev) [inline README.rst]: That might be too verbose. 
Consider adding description to each folder instead: backend, infra, tests. That would bring some clarity and benefits for new users

File: README.rst [hunk 2]
```diff
@@ -37,20 +44,44 @@ Installation
...
 2. Initialize and install dependencies using Poetry.
 
+    cd backend
 
+    poetry install
+
+3. Navigate to the ``infra`` directory.
+4. Initialize and install dependencies using Poetry.
+
+    cd ../infra
```
Comment (anatolyburtsev) [inline README.rst:42]: Is that step necessary given p.3 Navigate to infra directory?

File: backend/main.py [hunk 3]
```diff
@@ -1,14 +1,54 @@
...
+            "body": body_text,
+            "headers": {"content-type": "application/json"},
+        }
 
 
-@app.post("/reverse/")
-async def reverse_text(item: TextModel):
-    return {"reversed": item.text[::-1]}
+def validate_request(request_body):
+    if 'operation' not in request_body:
```
Comment (anatolyburtsev) [inline backend/main.py:17]: That super cool to have input request validation!
Please take a look to `Pydantic`, library, designed to do similar thing. Totally fine to add it in one of the following PRs (not now). Example how to use it: https://chat.openai.com/c/3e90a0b4-f9d0-4109-9917-19609b1c3e65

File: README.rst (hunk 2 above)
Comment (anya-mb) [reply README.rst:42]: I added it to make it more aligned with steps 1 and 2

File: README.rst (hunk 1 above)
Comment (anya-mb) [reply README.rst]: fixed it

File: README.rst (hunk 2 above)
Comment (anya-mb) [reply README.rst:42]: Changed it to step by step representation

File: tests/test_reverse_text_work_correctly.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
+import json
+from http import HTTPStatus
+import pytest
+from backend.main import lambda_reverse_text_backend
+
+
+def test_lambda_reverse_text_backend():
```
Comment (anatolyburtsev) [inline tests/test_reverse_text_work_correctly.py:7]: Would be cool to parametrized these tests, given they are super similar to each other

Comment (anya-mb) [reply tests/test_reverse_text_work_correctly.py:7]: Done!

**Review by anatolyburtsev — APPROVED** [review]
LGTM

**Pull request merged**